### ISSUE TITLE
Validate queue options and exit if option is not acceptable

### ISSUE THREAD

**Issue by silviucpp** [fogfish/esq#5]
Hello,

Having the following queue:

```erlang
{ok, Q}= esq:new( "priv/client_producer", [{tts, 0}, {capacity, 0}]).
```

After I add few items inside the queue and restart the app and do:

```erlang
esq:head(Q).
```
I always receive `undefined` but esq:deq always returns items. I'm expecting head to return same stuff as deq but without removing the item from queue. 

I'm using ` [{tts, 0}, {capacity, 0}]` to make sure all items are going on the disk directly. I'm wrong ?

Silviu

**Comment by fogfish** [comment]
I'll check it... but please keep in mind that `tts` is queue time-to-sync (rotate) file segments in milliseconds. I would not keep it equal to 0.

**Comment by silviucpp** [comment]
ohh ok :) thanks

**Comment by fogfish** [comment]
Right, I've checked your case deeply. Actually, `0` is not valid value for `tts` and `capacity`. The `capacity` defines the size of in-memory queue. The queue would not fetch anything from disk into memory buffer if capacity is `0`. You should not afraid to put capacity `> 0`. The queue always writes data to disk during enqueue.

**Comment by fogfish** [comment]
I would do the following improvements
- [x] Update README.md. The following statement is not valid: 'Queue starts to "swap" to file segments only when enqueue rate causes an overflow of in-memory buffer.'
- [ ] Accept `capacity` only if it is `> 0`
- [ ] Accept `tts` only if it is `> 0`

**Comment by silviucpp** [comment]
@fogfish you can also add documentation for `capacity` in the readme
